### PR TITLE
move scaleClient from factory

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Moves the ScaleClient from factory

cc @soltysh @deads2k

### CODE REVIEW

File: pkg/kubectl/cmd/apply.go [hunk 1]
```diff
@@ -223,7 +224,7 @@ func (o *ApplyOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 		return err
 	}
 
-	o.Scaler, err = f.ScaleClient()
+	o.Scaler, err = polymorphichelpers.ScaleClientFn(f)
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply.go]: This is  not polymorphic

File: pkg/kubectl/polymorphichelpers/interface.go [hunk 2]
```diff
@@ -113,3 +114,9 @@ type RollbackerFunc func(restClientGetter genericclioptions.RESTClientGetter, ma
 
 // RollbackerFn gives a way to easily override the function for unit testing if needed
 var RollbackerFn RollbackerFunc = rollbacker
+
+// ScaleClientFunc gives a way to retrieve a scale getter
+type ScaleClientFunc func(restClientGetter genericclioptions.RESTClientGetter) (scaleclient.ScalesGetter, error)
```
Comment (deads2k) [inline pkg/kubectl/polymorphichelpers/interface.go]: Make it a regular helper in cmd/util and we're good.

File: pkg/kubectl/cmd/apply.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/golang/glog"
 	"github.com/jonboulle/clockwork"
 	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl/polymorphichelpers"
```
Comment (wgliang) [inline pkg/kubectl/cmd/apply.go]: `goimports` the imported packages or manually sort.

File: pkg/kubectl/polymorphichelpers/interface.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/kubectl/polymorphichelpers/interface.go]: Thanks, done

File: pkg/kubectl/cmd/apply.go (hunk 3 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/apply.go]: Thanks, done

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

Although you might need to rebase this

**Comment by soltysh** [comment]
Closed in favor of https://github.com/kubernetes/kubernetes/pull/64374

**Pull request closed without merging**